### PR TITLE
Add the recommendation algorithm

### PR DESCRIPTION
**Pull request by BelfDev**
# Movie Recommendation Algorithm v1

- Fix dependency management issue
- Implement the movie recommendation algorithm
- Create the `api/recommendations/generate` endpoint (POST)
- Integrate the app with external TMDB api
- Integrate the app with external UTELLY api
- Adjust models

### CODE REVIEW

File: pickee_api/gateway.py [hunk 1]
```diff
@@ -27,71 +31,202 @@ def utelly_example_endpoint(request):
...
+
+        # Combines the genres selected as casual preferences with the ones marked as "favorites"
+        combined_genre_ids = __get_combined_genre_ids(casual_genre_ids, common_favorite_genres)
+        # Joins the combined genres separating the values with a comma
+        genre_string = ','.join(combined_genre_ids)
+
+        # Retrieves the favorite actors shared between all users of the session
+        common_favorite_actors = FavoriteFilter(model=FavoriteActor).get_common(users=users)
+        # Joins the common actors with a pipe character to indicate "OR" logic
+        actor_string = '|'.join(list(map(str, common_favorite_actors)))
```
Comment (rjs68) [inline pickee_api/gateway.py]: Does this join actor ids?

File: pickee_api/gateway.py [hunk 2]
```diff
@@ -27,71 +31,202 @@ def utelly_example_endpoint(request):
...
+
+        # Retrieves the favorite movies shared between all users of the session
+        common_favorite_movies = FavoriteFilter(model=FavoriteMovie).get_common(users=users)
+
+        # Retrieves the keywords associated with the common movies
+        keywords = __get_movie_keywords(common_favorite_movies)
+        keyword_string = '|'.join(list(map(str, keywords)))
```
Comment (rjs68) [inline pickee_api/gateway.py]: Keyword ids?

File: pickee_api/gateway.py [hunk 3]
```diff
@@ -27,71 +31,202 @@ def utelly_example_endpoint(request):
...
+        # Infers the TMDB page based on the offset
+        page = (offset // TMDB_ITEMS_PER_PAGE) + 1
+
+        # Builds the request URL
+        query_params = {
+            'language': 'en-UK',
+            'sort_by': 'popularity.desc',
+            'page': page,
+            'with_genres': genre_string,
+            'with_actors': actor_string,
```
Comment (rjs68) [inline pickee_api/gateway.py]: with_cast

File: pickee_api/gateway.py (hunk 1 above)
Comment (BelfDev) [reply pickee_api/gateway.py]: yeah

File: pickee_api/gateway.py (hunk 2 above)
Comment (BelfDev) [reply pickee_api/gateway.py]: yeah

File: pickee_api/gateway.py (hunk 3 above)
Comment (BelfDev) [reply pickee_api/gateway.py]: thanks Rhys!

**Pull request merged**